### PR TITLE
Shear fields vector

### PR DESCRIPTION
**Pull request by tberlok**

### CODE REVIEW

**Comment by tberlok** [comment]
These three commits add some needed functionality for studying the current in shear experiments.

File: skeletor/cython/deposit.pyx [hunk 1]
```diff
@@ -15,7 +15,7 @@ def deposit(
         x = particles[ip].x
         y = particles[ip].y
 
-        vx = particles[ip].vx
```
Comment (tobson) [inline skeletor/cython/deposit.pyx:18]: Maybe add a comment that this is the relative or fluctuating velocity.

File: skeletor/field.py [hunk 2]
```diff
@@ -294,6 +335,25 @@ def translate(self, time):
         # Set boundary condition?
         self.boundaries_set = False
 
+    def translate_vector(self, time):
+        """Translation using numpy's fft."""
+        from numpy.fft import rfft, irfft
+        from numpy import exp
+
+        dims = ('x', 'y')
```
Comment (tobson) [inline skeletor/field.py]: Combine this line and the following one.

File: skeletor/sources.py [hunk 3]
```diff
@@ -1,26 +1,34 @@
...
 
 class Sources:
 
-    def __init__(self, grid, **kwds):
+    def __init__(self, manifold, **kwds):
 
-        self.rho = Field(grid, dtype=Float, **kwds)
-
-        self.J = Field(grid, dtype=Float2, **kwds)
+        if not manifold.shear:
```
Comment (tobson) [inline skeletor/sources.py:10]: In order to reduce the amount of redundant code, wouldn't it be possible to say

```python
if manifold.shear:
  from .field import ShearField as Field
else:
  from .field import Field

self.rho = Field(manifold, dtype=Float, **kwds)
self.J = Field(manifold, dtype=Float2, **kwds)
```

?

File: skeletor/sources.py [hunk 4]
```diff
@@ -1,26 +1,34 @@
...
     def deposit(self, particles, erase=True):
 
         if erase:
             self.rho.fill(0.0)
             self.J.fill((0.0, 0.0))
 
-        cython_deposit(
-                particles[:particles.np], self.rho, self.J, particles.charge,
-                self.rho.grid.noff, self.rho.grid.lbx, self.rho.grid.lby)
+        if not self.rho.grid.shear:
```
Comment (tobson) [inline skeletor/sources.py:24]: Is there a reason for the use of `elif shear` instead of `else`?

**Comment by tobson** [comment]
I love this. I think short-lived branches plus code review is a great way of keeping me in the loop.

File: skeletor/cython/deposit.pyx (hunk 1 above)
Comment (tberlok) [reply skeletor/cython/deposit.pyx:18]: Done

File: skeletor/sources.py (hunk 4 above)
Comment (tberlok) [reply skeletor/sources.py:24]: I think I initially meant to put a final `else` as a sanity check but I think it is not necessary.

**Pull request merged**